lib/chef/provider/package/rubygems.rb

```diff
@@ -469,7 +470,8 @@ def all_installed_versions
         end
 
         def gem_sources
-          new_resource.source ? Array(new_resource.source) : nil
+          srcs = new_resource.source || Chef::Config[:rubygems_url]
```

Comment:
it seems like this also fixes a bug where gem_package installs without a binary shell_out have never used the config variable.